src/main/java/org/testng/SuiteRunner.java

```diff
@@ -220,9 +247,17 @@ public int compare(XmlTest arg0, XmlTest arg1) {
       }
     });
 
+    boolean isNewVariant = (iTestRunnerFactory instanceof ITestRunnerFactory2);
```

Comment:
In fact, it should always be `ITestRunnerFactory2`. But we should have a proxy `ITestRunnerFactory2` impl which contains a `ITestRunnerFactory` and call the appropriate method.